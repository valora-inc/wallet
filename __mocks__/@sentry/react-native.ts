const Sentry = {
  config: () => ({ install: jest.fn() }),
  setTagsContext: jest.fn(),
  captureException: jest.fn(),
  captureMessage: jest.fn(),
  ReactNavigationInstrumentation: jest.fn().mockImplementation(() => ({})),
  Severity: {
    Error: 'error',
  },
  startTransaction: jest.fn(),
  nativeCrash: jest.fn(),
  wrap: jest.fn().mockImplementation((x) => x),
}

module.exports = Sentry
