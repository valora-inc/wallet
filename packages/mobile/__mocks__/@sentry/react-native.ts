const Sentry = {
  config: () => ({ install: jest.fn() }),
  setTagsContext: jest.fn(),
  captureException: jest.fn(),
  captureMessage: jest.fn(),
  ReactNavigationInstrumentation: jest.fn().mockImplementation(() => ({})),
  Severity: {
    Error: 'error',
  },
}

module.exports = Sentry
