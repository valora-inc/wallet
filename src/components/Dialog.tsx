import * as React from 'react'
import {
  ActivityIndicator,
  Image,
  ImageSourcePropType,
  ScrollView,
  StyleSheet,
  Text,
  View,
} from 'react-native'
import Modal from 'src/components/Modal'
import TextButton from 'src/components/TextButton'
import colors from 'src/styles/colors'
import fontStyles from 'src/styles/fonts'

interface Props {
  image?: ImageSourcePropType
  title?: string | React.ReactNode
  children: React.ReactNode
  actionText?: string
  isActionHighlighted?: boolean
  actionPress?: () => void
  secondaryActionText?: string
  secondaryActionDisabled?: boolean
  secondaryActionPress?: () => void
  isVisible: boolean
  showLoading?: boolean
  testID?: string
  onBackgroundPress?: () => void
}

export default function Dialog({
  title,
  children,
  actionPress,
  actionText,
  isActionHighlighted = true,
  secondaryActionText,
  secondaryActionDisabled,
  secondaryActionPress,
  showLoading = false,
  image,
  isVisible,
  testID,
  onBackgroundPress,
}: Props) {
  return (
    <Modal isVisible={isVisible} testID={testID} onBackgroundPress={onBackgroundPress}>
      <ScrollView contentContainerStyle={styles.root}>
        {image && <Image style={styles.imageContainer} source={image} resizeMode="contain" />}
        {title && <Text style={styles.title}>{title}</Text>}
        <Text style={styles.body}>{children}</Text>
      </ScrollView>
      <View style={styles.actions}>
        {secondaryActionText && (
          <TextButton
            style={styles.secondary}
            disabled={secondaryActionDisabled}
            onPress={secondaryActionPress}
            testID={testID && `${testID}/SecondaryAction`}
          >
            {secondaryActionText}
          </TextButton>
        )}
        {showLoading ? (
          <ActivityIndicator style={styles.primary} size="small" color={colors.greenUI} />
        ) : (
          <>
            {actionText && (
              <TextButton
                style={isActionHighlighted ? styles.primary : styles.secondary}
                onPress={actionPress}
                testID={testID && `${testID}/PrimaryAction`}
              >
                {actionText}
              </TextButton>
            )}
          </>
        )}
      </View>
    </Modal>
  )
}

const styles = StyleSheet.create({
  root: {
    alignItems: 'center',
  },
  title: {
    textAlign: 'center',
    marginBottom: 12,
    ...fontStyles.h2,
  },
  body: {
    textAlign: 'center',
    ...fontStyles.regular,
    marginBottom: 24,
  },
  actions: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    maxWidth: '100%',
    flexWrap: 'wrap',
  },
  secondary: {
    color: colors.gray4,
    paddingTop: 16,
  },
  primary: {
    paddingTop: 16,
  },
  imageContainer: {
    marginBottom: 12,
    width: 100,
    height: 100,
  },
})
