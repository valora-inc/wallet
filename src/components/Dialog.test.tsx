import { render } from '@testing-library/react-native'
import * as React from 'react'
import 'react-native'
import Dialog from 'src/components/Dialog'

it('renders correctly', () => {
  const onPress = jest.fn()
  const tree = render(
    <Dialog isVisible={true} title="Dialog" actionPress={onPress} actionText="Press Me">
      "HELLO"
    </Dialog>
  )
  expect(tree).toMatchSnapshot()
})
