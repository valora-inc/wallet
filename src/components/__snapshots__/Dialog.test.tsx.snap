// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="none"
  deviceHeight={1334}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={true}
  supportedOrientations={
    [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={1334}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={true}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": [
          {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <RNCSafeAreaView
      edges={
        [
          "bottom",
          "left",
          "right",
          "top",
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#FFFFFF",
              "padding": 16,
            },
            {
              "borderRadius": 8,
            },
            {
              "elevation": 12,
              "shadowColor": "rgba(156, 164, 169, 0.4)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 12,
            },
            [
              {
                "backgroundColor": "#FFFFFF",
                "maxHeight": "100%",
                "padding": 24,
              },
              undefined,
            ],
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            {
              "alignItems": "center",
            }
          }
        >
          <View>
            <Text
              style={
                {
                  "color": "#2E3338",
                  "fontFamily": "Jost-Medium",
                  "fontSize": 22,
                  "lineHeight": 28,
                  "marginBottom": 12,
                  "textAlign": "center",
                }
              }
            >
              Dialog
            </Text>
            <Text
              style={
                {
                  "color": "#2E3338",
                  "fontFamily": "Inter-Regular",
                  "fontSize": 16,
                  "lineHeight": 22,
                  "marginBottom": 24,
                  "textAlign": "center",
                }
              }
            >
              "HELLO"
            </Text>
          </View>
        </RCTScrollView>
        <View
          style={
            {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-around",
              "maxWidth": "100%",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            nativeBackgroundAndroid={
              {
                "attribute": "selectableItemBackgroundBorderless",
                "rippleRadius": undefined,
                "type": "ThemeAttrAndroid",
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text
              style={
                [
                  {
                    "color": "#1AB775",
                    "fontFamily": "Inter-SemiBold",
                    "fontSize": 16,
                    "lineHeight": 22,
                  },
                  {
                    "paddingTop": 16,
                  },
                ]
              }
            >
              Press Me
            </Text>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </View>
</Modal>
`;
